### ISSUE TITLE
Warnings cannot be turned off for batch testing

### ISSUE THREAD

**Issue by multiflexi** [drwetter/testssl.sh#2496]
I am running version commit e3b3c358fd03c20d251096b7e845d90861f41dde

**Command line to reproduce**
```
export WARNINGS=off
./testssl.sh -iL list
```
**Expected behavior**
Ignore warnings when server supports only TLS 1.3, but this is the output:
```
====== Running in file batch mode with file="list" ======

=========================================================================================================
./testssl.sh --warnings=batch sner.flab.cesnet.cz
```

When  `./testssl.sh -iL list --warnings off ` is used, this is the output:
```
====== Running in file batch mode with file="list" ======

=========================================================================================================
./testssl.sh --warnings=batch --warnings off sner.flab.cesnet.cz
```

**Your system (please complete the following information):**
 - OS: `Manjaro Linux`
 - Platform: `Linux 6.9.1-1-MANJARO x86_64` 
 - OpenSSL: `./bin/openssl.Linux.x86_64`

**Comment by drwetter** [comment]
Hi @multiflexi ,

not sure I can follow you. Are you implicating that _====== Running in file batch mode with file="list" ======_ is a warning? If so: it's not . It's just a message reflecting what you instructed testssl.sh to do.

**Comment by multiflexi** [comment]
No, this part is important: `./testssl.sh --warnings=batch sner.flab.cesnet.cz` vs `./testssl.sh --warnings=batch --warnings off sner.flab.cesnet.cz`. It is impossible to turn off warnings with environment variable, so when server supports only TLS 1.3, you still get warning requiring user input, therefore it does not test the server.

**Comment by drwetter** [comment]
Guess I should then rather test it on a system which provides the binary and not MacOS 😬

**Comment by drwetter** [comment]
Okay. Upfront: The process regarding TLS 1.3 only-hosts will be improved, this has been discussed.

I understand the logic regarding warnings it's currently a bit cumbersome.

* ``./testssl.sh -iL list`` automatically sets WARNINGS to batch, see man page. It doesn´t matter if ``export WARNINGS=off`` is supplied.
* ``./testssl.sh --warnings=batch --warnings off sner.flab.cesnet.cz`` uses the last key/value pair. This a user error we currently don't deal with. 

The first one would be a less than a one-liner to fix, like ``[[ "$WARNINGS" != off ]] && WARNINGS="batch"`` around L23588.

**Comment by multiflexi** [comment]
Why is WARNINGS set to batch automatically? I think it is more logical to not show any warnings when batch scanning is used, because this is usually used without user interaction. Also it is a bit confusing if the documentation states for --warnings that "The same can be achieved by setting the environment variable WARNINGS."

**Comment by drwetter** [comment]
well, warnings have a reason. That's what they are for.

> --warnings that "The same can be achieved by setting the environment variable WARNINGS."

That was the wrong section. the right one is:  ``--file <fname> or the equivalent -iL <fname> are mass testing options. Per default it implicitly turns on --warnings batch.``

**Comment by multiflexi** [comment]
I understand that warnings have a reason, but batch scanning also. You use it so you do not have to punch in each URL individually. And then maybe get a tea. But when you have to babysit the batch process, it kinda looses its purpose IMHO.

**Comment by drwetter** [comment]
The pending PR now gives you the possibility to overwrite the implicit ``WARNINGS=batch`` when using ``--files``, so that ``export WARNINGS=off ./testssl.sh -iL list`` works.

Other than that as indicated I am not 100% happy with this parameter either. But a) that will be addressed sometime later b) the TLS 1.3-only-hosts is a separate issue, see e.g.  #1312
